app/scripts/controllers/createPersistentVolumeClaim.js

```diff
@@ -55,29 +55,36 @@ angular.module('openshiftConsole')
           }
         };
 
-      function generatePersistentVolumeClaim() {
-        var pvc = {
-          kind: "PersistentVolumeClaim",
-          apiVersion: "v1",
-          metadata: {
-            name: $scope.claim.name,
-            labels: {}
-          },
-          spec: {
-            resources: {
-              requests:{}
+         function generatePersistentVolumeClaim() {
+          var pvc = {
+            kind: "PersistentVolumeClaim",
+            apiVersion: "v1",
+            metadata: {
+              name: $scope.claim.name,
+              labels: {}
+            },
+            spec: {
+              resources: {
+                requests:{}
 
+              },
+              selector:{
+                matchLabels:{}
+              }
             }
-          }
-        };
-
-        pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
-        var unit =  $scope.claim.unit || "Mi";
-        pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
-
-        return pvc;
-      }
+          };
 
+          pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
+          var unit =  $scope.claim.unit || "Mi";
+          pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
+          if ($scope.claim.selectedLabels.length > 0)
```

Comment:
The length check is unnecessary. `_.each` handles empty arrays (and even `undefined` or `null`).